File: ground_truths/geeks_for_geeks_successful_test_scripts/python/FIND_RECTANGLE_BINARY_MATRIX_CORNERS_1_1.py
def f_gold ( matrix ) :
	rows = len ( matrix )
	if ( rows == 0 ) :
		return False
	columns = len ( matrix [ 0 ] )
	table = { }
	for i in range ( rows ) :
		for j in range ( columns - 1 ) :
			for k in range ( j + 1 , columns ) :
				if ( matrix [ i ] [ j ] == 1 and matrix [ i ] [j -1]):
					if ( j in table and k in table [ j ] ) :
						return True
					if ( k in table and j in table [ k ] ) :
						return True
					if j not in table :
						table [ j ] = set ( )
					if k not in table :
						table [ k ] = set ( )
						table [ j ].add ( k )
						table [ k ].add ( j )
	return False

File: results/java/FIND_RECTANGLE_BINARY_MATRIX_CORNERS_1_1.cpp
bool f_filled ( vector < vector < int >> matrix ) {
  int rows = matrix . size ( ) ;
  if ( rows == 0 ) return false ;
  int columns = matrix [ 0 ] . size ( ) ;
  unordered_map < int , set < int >> table ;
  for ( int i = 0 ;
  i < rows ;
  i ++ ) {
    for ( int j = 0 ;
    j < columns - 1 ;
    j ++ ) {
      for ( int k = j + 1 ;
      k < columns ;
      k ++ ) {
        if ( matrix [ i ] [ j ] == 1 && matrix [ i ] [ k ] == 1 ) {
          if ( table . find ( j ) != table . end ( ) && table [ j ] . find ( k ) != table . end ( ) ) return true ;
          if ( table . find ( k ) != table . end ( ) && table [ k ] . find ( j ) != table . end ( ) ) return true ;
          if ( ! table . find ( j ) != table . end ( ) ) {
            set < int > x ;
            x . insert ( k ) ;
            table [ j ] . insert ( x ) ;
          }
          else table [ j ] . insert ( k ) ;
          if ( ! table . find ( k ) != table . end ( ) ) {
            set < int > x ;
            x . insert ( j ) ;
            table [ k ] . insert ( x ) ;
          }
          else table [ k ] . insert ( j ) ;
        }
      }
    }
  }
  return false ;
}
